src/components/app/list-new/GenericAppList.tsx

```diff
@@ -101,184 +135,111 @@ const GenericAppList = ({
+        isArgoCDAppList ? handleArgoAppListing() : handleFluxAppListing()
     }, [clusterIdsCsv])
 
-    // reset data
-    function init() {
-        if (isSSE && !payloadParsedFromUrl?.namespaces?.length) {
-            setDataStateType(AppListViewType.LIST)
-        } else {
-            setDataStateType(AppListViewType.LOADING)
-        }
-        setClusterIdsCsv(_getClusterIdsFromRequestUrl() ?? null)
+    const init = () => {
+        clusterIdsCsv ? setDataStateType(AppListViewType.LOADING) : setDataStateType(AppListViewType.LIST)
         setAppsList([])
-        setFilteredAppsList([])
         if (sseConnection) {
             sseConnection.close()
         }
         setSseConnection(null)
     }
 
-    function _getClusterIdsFromRequestUrl() {
-        return [...buildClusterVsNamespace(payloadParsedFromUrl.namespaces?.join(',')).keys()].join(',')
-    }
-
-    function handleFilteration() {
-        const _search = payloadParsedFromUrl.appNameSearch
-        const _sortOrder = payloadParsedFromUrl.sortOrder
-        const selectedTemplateTypes = payloadParsedFromUrl.templateType || []
-        let _filteredAppsList = [...(appsList || [])]
-
-        // handle search
-        if (_search?.length) {
-            const searchLowerCase = _search.toLowerCase()
-            _filteredAppsList = _filteredAppsList.filter((app) => app.appName.toLowerCase().includes(searchLowerCase))
-        }
-
-        // handle sorting by ascending/descending order
-        if (_sortOrder == OrderBy.ASC) {
-            _filteredAppsList = _filteredAppsList.sort((a, b) => a.appName.localeCompare(b.appName))
-        } else {
-            _filteredAppsList = _filteredAppsList.sort((a, b) => b.appName.localeCompare(a.appName))
-        }
-
-        // handle template type filter
-        if (selectedTemplateTypes.length > 0) {
-            _filteredAppsList = _filteredAppsList.filter((app) =>
-                selectedTemplateTypes.includes(app.fluxAppDeploymentType),
-            )
-        }
+    const handleAppNameSorting = () => handleSorting(AppListSortableKeys.APP_NAME)
 
-        setSortOrder(_sortOrder)
-        setFilteredAppsList(_filteredAppsList)
-        setShowPulsatingDotState(_filteredAppsList.length == 0 && !clusterIdsCsv)
-    }
+    const isAnyFilterAppliedExceptCluster = !!searchKey || !!templateType.length || !!namespace.length
 
-    function _isAnyFilterationAppliedExceptClusterAndNs() {
-        return (
-            payloadParsedFromUrl.teams?.length ||
-            payloadParsedFromUrl.appNameSearch?.length ||
-            payloadParsedFromUrl.environments?.length ||
-            payloadParsedFromUrl.templateType?.length
-        )
-    }
+    const isAnyFilterApplied = isAnyFilterAppliedExceptCluster || !!clusterIdsCsv
 
-    function _isAnyFilterationApplied() {
-        return _isAnyFilterationAppliedExceptClusterAndNs() || !!clusterIdsCsv
-    }
+    const isAllClusterSelected = cluster.length === clusterList?.length
```

Comment:
Check for ? check

Reply:
Can we just send the length from parent instead of whole list, if we are not utilizing it